core/modules/memman/MemFileSet.h

```diff
@@ -0,0 +1,120 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_QSERV_MEMMAN_MEMFILESET_H
+#define LSST_QSERV_MEMMAN_MEMFILESET_H
+
+// System headers
+#include <string>
+#include <unistd.h>
```

Comment:
This should not happen. On any platform `size_t` is guaranteed to be at least the size of pointer (size_t can be used to index any array/container type and the size of the container could be the whole virtual memory). I'd be very surprised to see different definitions in different headers, it should all come from some kernel-level typedef.